pkg/yaml/secure/secure.go

```diff
@@ -1,79 +1,36 @@
 package secure
 
 import (
-	"crypto/aes"
-	"crypto/cipher"
-	"crypto/rand"
-	"encoding/base64"
-	"fmt"
-	"io"
+	"crypto/sha256"
+	"hash"
 
 	"github.com/drone/drone/Godeps/_workspace/src/gopkg.in/yaml.v2"
+
+	common "github.com/drone/drone/pkg/types"
+	"github.com/drone/drone/pkg/utils/sshutil"
 )
 
 // Parse parses and returns the secure section of the
 // yaml file as plaintext parameters.
-func Parse(key, raw string) (map[string]string, error) {
+func Parse(repo *common.Repo, raw string) (map[string]string, error) {
```

Comment:
since the data model has changed a bit (and may still change) I'd like to avoid sending the repo model to the `secure` package. Do you think we could do this instead?

```diff
-func Parse(repo *common.Repo, raw string)
+func Parse(privateKey, hash, raw string)
```